project-koku/koku
# [COST-5367] - Fix Azure ingress required columns

## Jira Ticket

[COST-5367](https://issues.redhat.com/browse/COST-5367)

## Description

This change will update the required columns for the Azure customer filtered data flow. Azure report columns come in many different formats WRT case sensitivity. This change consolidates the required columns and validates them all in lower case. Additionally this goes hand in hand with the script updates for RHEL metering in https://github.com/project-koku/koku-data-selector/pull/39/ that PR calls out the case restriction and updates the script to normalise them.

## Testing

1. Checkout Branch
2. Restart Koku
3. Add storage only Azure source (customer filtered flow) [Luke can help here, there is also some helpful scripts on Inscope]
4. Send an Azure ingress payload with required columns
5. Processing should complete successfully
6. Send a second Azure payload with missing columns
7. Should see an error message raised with missing columns listed.

## Release Notes

```markdown
* [COST-5367](https://issues.redhat.com/browse/COST-5367) Fix Azure ingress required columns
```


## Notes

This may require some IQE test updates specifically `test_api_azure_hcs_customer_filtered_report`
